File: truck/ECC/ZZB/ecc/doc/QueueDaoImpl.cs
using System;
using System.Collections.Generic;
using System.Text;

using SiteView.Ecc.Core.Dao;
using SiteView.Ecc.WSClient.XFireWebService;
using SiteView.Ecc.Core;
using SiteView.Ecc.Core.Models;
using System.Collections.Specialized;

namespace SiteView.Ecc.WSClient
{
    public class QueueDaoImpl : IQueueDao
    {

        #region IQueueDao ≥…‘±
        private InterfaceEccService service = new InterfaceEccService();
        private static log4net.ILog logger = log4net.LogManager.GetLogger(typeof(QueueDaoImpl));
        
        public void CreateQueue(string queuename)
        {
            this.service.Url = UserPermissionContext.Instance.Url;
            anyType2anyTypeMapEntry[] requestData = new anyType2anyTypeMapEntry[]
            {
                Helper.XfireCreateKeyValue("dowhat","CreateQueue"),
                Helper.XfireCreateKeyValue("queuename",queuename),
            };
            RetMapInVector result = ServiceClient.GetUnivData2(requestData);
            if (!result.retbool)
            {
                if (logger.IsWarnEnabled)
                {
                    logger.Warn(result.estr);
                }
            }
            IDictionary<string, NameValueCollection> resultData = Helper.XfireConventMapEntryToDictionary(result.vmap);

        }

        public bool PushStringMessage(string queuename, string label, string content)
        {
            this.service.Url = UserPermissionContext.Instance.Url;
            anyType2anyTypeMapEntry[] requestData = new anyType2anyTypeMapEntry[]
            {
                Helper.XfireCreateKeyValue("dowhat","PushStringMessage"),
                Helper.XfireCreateKeyValue("queuename",queuename),
                Helper.XfireCreateKeyValue("label",label),
                Helper.XfireCreateKeyValue("content",content)
            };
            RetMapInVector result = ServiceClient.GetUnivData2(requestData);
            bool rs = false;
            if (result.retbool)
            {
                rs = true;
            }
            else
            {
                if (logger.IsWarnEnabled)
                {
                    logger.Warn(result.estr);
                }
            }
            IDictionary<string, NameValueCollection> resultData = Helper.XfireConventMapEntryToDictionary(result.vmap);
            return rs;

        }

        public void GetMQRecordCount(string queuename)
        {
            this.service.Url = UserPermissionContext.Instance.Url;
            anyType2anyTypeMapEntry[] requestData = new anyType2anyTypeMapEntry[]
            {
                Helper.XfireCreateKeyValue("dowhat","GetMQRecordCount"),
                Helper.XfireCreateKeyValue("queuename",queuename),
            };
            RetMapInVector result = ServiceClient.GetUnivData2(requestData);
            if (!result.retbool)
            {
                if (logger.IsWarnEnabled)
                {
                    logger.Warn(result.estr);
                }
            }
            IDictionary<string, NameValueCollection> resultData = Helper.XfireConventMapEntryToDictionary(result.vmap);

        }

        public void GetAllQueueNames()
        {
            this.service.Url = UserPermissionContext.Instance.Url;
            anyType2anyTypeMapEntry[] requestData = new anyType2anyTypeMapEntry[]
            {
                Helper.XfireCreateKeyValue("dowhat","GetAllQueueNames"),
            };
            RetMapInVector result = ServiceClient.GetUnivData2(requestData);
            if (!result.retbool)
            {
                if (logger.IsWarnEnabled)
                {
                    logger.Warn(result.estr);
                }
            }
            IDictionary<string, NameValueCollection> resultData = Helper.XfireConventMapEntryToDictionary(result.vmap);
        }

        public void DeleteQueue(string queuename)
        {
            this.service.Url = UserPermissionContext.Instance.Url;
            anyType2anyTypeMapEntry[] requestData = new anyType2anyTypeMapEntry[]
            {
                Helper.XfireCreateKeyValue("dowhat","DeleteQueue"),
                Helper.XfireCreateKeyValue("queuename",queuename),
            };
            RetMapInVector result = ServiceClient.GetUnivData2(requestData);
            if (!result.retbool)
            {
                if (logger.IsWarnEnabled)
                {
                    logger.Warn(result.estr);
                }
            }
            IDictionary<string, NameValueCollection> resultData = Helper.XfireConventMapEntryToDictionary(result.vmap);
        }

        public void ClearQueueMessage(string queuename)
        {
            this.service.Url = UserPermissionContext.Instance.Url;
            anyType2anyTypeMapEntry[] requestData = new anyType2anyTypeMapEntry[]
            {
                Helper.XfireCreateKeyValue("dowhat","ClearQueueMessage"),
                Helper.XfireCreateKeyValue("queuename",queuename),
            };
            RetMapInVector result = ServiceClient.GetUnivData2(requestData);
            if (!result.retbool)
            {
                if (logger.IsWarnEnabled)
                {
                    logger.Warn(result.estr);
                }
            }
            IDictionary<string, NameValueCollection> resultData = Helper.XfireConventMapEntryToDictionary(result.vmap);
        }
       
        #endregion


        
    }
        
}

File: truck/ECC/ZZB/ecc/src/com/siteview/ecc/alert/dao/QueueDaoImpl.java
package com.siteview.ecc.alert.dao;

import java.util.HashMap;

import com.siteview.base.manage.ManageSvapi;
import com.siteview.base.manage.RetMapInMap;

public class QueueDaoImpl implements IQueueDao {

	@Override
	public void clearQueueMessage(String queuename) throws Exception{
		HashMap<String, String> ndata = new HashMap<String, String>();
		ndata.put("dowhat", "ClearQueueMessage");
		ndata.put("queuename", queuename);
		RetMapInMap ret = ManageSvapi.GetUnivData(ndata);
		if (!ret.getRetbool())
			throw new Exception(ret.getEstr());
	}

	@Override
	public void createQueue(String queuename) throws Exception {
		HashMap<String, String> ndata = new HashMap<String, String>();
		ndata.put("dowhat", "CreateQueue");
		ndata.put("queuename", queuename);
		RetMapInMap ret = ManageSvapi.GetUnivData(ndata);
		if (!ret.getRetbool())
			throw new Exception("Failed to createQueue :" + ret.getEstr());
	}

	@Override
	public void deleteQueue(String queuename) throws Exception{
		HashMap<String, String> ndata = new HashMap<String, String>();
		ndata.put("dowhat", "DeleteQueue");
		ndata.put("queuename", queuename);
		RetMapInMap ret = ManageSvapi.GetUnivData(ndata);
		if (!ret.getRetbool())
			throw new Exception(ret.getEstr());
	}

	@Override
	public void getAllQueueNames() throws Exception{
		HashMap<String, String> ndata = new HashMap<String, String>();
		ndata.put("dowhat", "GetAllQueueNames");
		RetMapInMap ret = ManageSvapi.GetUnivData(ndata);
		if (!ret.getRetbool())
			throw new Exception(ret.getEstr());
	}

	@Override
	public void getMQRecordCount(String queuename)throws Exception {
		HashMap<String, String> ndata = new HashMap<String, String>();
		ndata.put("dowhat", "GetMQRecordCount");
		ndata.put("queuename", queuename);
		RetMapInMap ret = ManageSvapi.GetUnivData(ndata);
		if (!ret.getRetbool())
			throw new Exception(ret.getEstr());
	}

	@Override
	public boolean pushStringMessage(String queuename, String label,
			String content) throws Exception{
		HashMap<String, String> ndata = new HashMap<String, String>();
		ndata.put("dowhat", "PushStringMessage");
		ndata.put("queuename", queuename);
		ndata.put("label", label);
		ndata.put("content", content);
		RetMapInMap ret = ManageSvapi.GetUnivData(ndata);
        return ret.getRetbool();
	}

}
